### PR TITLE
PLANET-3488 Save p4 enform fields in custom meta field

### PR DESCRIPTION
**Pull request by kirdia**
Combines PLANET-3488 & PLANET-3410 #99 .

Persist customized form fields to p4_enform meta field 'p4enform_fields'

### CODE REVIEW

File: admin/js/enforms.js [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+  /**
+   * Event listener for add field/question button.
+   */
+  $('.add-en-field').off('click').on('click', function (e) {
+    e.preventDefault();
+
+    $(this).prop('disabled', true);
+    var id = $(this).data('id');
+    var name = $(this).data('name');
+    var type = $(this).data('type');
```
Comment (dpivo) [inline admin/js/enforms.js]: phpcs error: equals signs not aligned with surrounding ones

File: classes/controller/menu/class-enform-post-controller.php [hunk 2]
```diff
@@ -318,6 +320,76 @@ public function display_optins_custom_box() {
...
+				],
+				'0.2',
+				true
+			);
+		}
+
+		/**
+		 * Saves the p4 enform fields of the Post.
+		 *
+		 * @param int $post_id The ID of the current Post.
```
Comment (dpivo) [inline classes/controller/menu/class-enform-post-controller.php]: phpcs error: parameter types not lining up

File: classes/controller/menu/class-enform-post-controller.php [hunk 3]
```diff
@@ -71,6 +72,12 @@ public function create_admin_menu() {
 					'post-new.php?post_type=' . self::POST_TYPE
 				);
 
+				// Load assets conditionally using pagenow, typenow on new form page.
+				global $pagenow, $typenow;
+				if ( 'post-new.php' === $pagenow && self::POST_TYPE === $typenow ) {
+					add_action( 'load-post-new.php', [ $this, 'load__new_page_assets' ] );
+					add_action( 'admin_print_footer_scripts', [ $this, 'print_admin_footer_scripts' ], 1 );
+				}
```
Comment (angtheod) [inline classes/controller/menu/class-enform-post-controller.php]: Can we add it also in the Edit Form page?

Comment (angtheod) [reply classes/controller/menu/class-enform-post-controller.php]: Ok, we 'll be added in next ticket

**Pull request merged**